src/map/pc.cpp

```diff
@@ -2804,19 +2804,19 @@ static void pc_bonus_addeff_onskill(std::vector<s_addeffectonskill> &effect, enu
 
 	for (auto &it : effect) {
 		if (it.sc == sc && it.skill_id == skill_id && it.target == target) {
-			it.rate = cap_value(it.rate + rate, -10000, 10000);
+			it.rate = util::safe_addition_cap(it.rate, rate, SHRT_MAX);
 			it.duration = umax(it.duration, duration);
 			return;
 		}
 	}
 
 	struct s_addeffectonskill entry = {};
 
-	if (rate < -10000 || rate > 10000)
-		ShowWarning("pc_bonus_addeff_onskill: Item bonus rate %d exceeds -10000~10000 range, capping.\n", rate);
+	if (rate < SHRT_MIN || rate > SHRT_MAX)
+		ShowWarning("pc_bonus_addeff_onskill: Item bonus rate %d exceeds -%d~%d range, capping.\n", rate, SHRT_MIN, SHRT_MAX);
 
 	entry.sc = sc;
-	entry.rate = cap_value(rate, -10000, 10000);
+	entry.rate = cap_value(rate, SHRT_MIN, SHRT_MAX);
```

Comment:
The cap seems to be only working for the first item.
For example I had two cards that gave a 40000 (400%) chance to freeze.
The resulting chance was 32767+40000=72767.